s3-import-commons/src/main/java/no/unit/nva/publication/s3imports/SqsBatchMessenger.java

```diff
@@ -0,0 +1,113 @@
+package no.unit.nva.publication.s3imports;
+
+import static nva.commons.core.attempt.Try.attempt;
+import com.amazonaws.services.sqs.AmazonSQS;
+import com.amazonaws.services.sqs.model.SendMessageBatchRequest;
+import com.amazonaws.services.sqs.model.SendMessageBatchRequestEntry;
+import com.amazonaws.services.sqs.model.SendMessageBatchResult;
+import com.amazonaws.services.sqs.model.SendMessageBatchResultEntry;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import no.unit.nva.events.models.EventReference;
+import no.unit.nva.identifiers.SortableIdentifier;
+import nva.commons.core.attempt.Failure;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class SqsBatchMessenger {
+
+    public static final int MAX_NUMBER_OF_MESSAGES_PER_BATCH_ALLOWED_BY_AWS = 10;
+    private static final Logger logger = LoggerFactory.getLogger(SqsBatchMessenger.class);
+    private final AmazonSQS client;
+    private final String queue;
+
+    private final Map<String, EventReference> requestIdToMessageBody;
+
+    public SqsBatchMessenger(AmazonSQS client, String queue) {
+        this.client = client;
+        this.queue = queue;
+        this.requestIdToMessageBody = new HashMap<>();
+    }
+
+    public PutSqsMessageResult sendMessages(List<EventReference> messageBodies) {
+        var messageRequestEntries = createMessageRequestEntries(messageBodies);
+        var batches = batchEventReferences(messageRequestEntries);
+        var batchRequest = batches.stream().map(this::mapToSendMessageBatchRequestEntry);
+        return batchRequest.map(this::sendToSqs).reduce(PutSqsMessageResult::combine).orElseThrow();
```

Comment:
Bare forslag:

return batches.stream().map(this::mapToSendMessageBatchRequestEntry)
                               .map(this::sendToSqs)
                               .reduce(PutSqsMessageResult::combine)
                               .orElseThrow();